src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -916,18 +930,38 @@ internal void Log(GrpcEvent msg)
                     }
                     else
                     {
-                        if (rpcLog.Exception != null)
+                        bool cleanup = false;
+                        try
                         {
-                            // TODO fix RpcException catch all https://github.com/Azure/azure-functions-dotnet-worker/issues/370
-                            var exception = new Workers.Rpc.RpcException(rpcLog.Message, rpcLog.Exception.Message, rpcLog.Exception.StackTrace);
-                            context.Logger.Log(logLevel, new EventId(0, rpcLog.EventId), rpcLog.Message, exception, (state, exc) => state);
+                            if (stateTuple.AppInsightsEnabledOnWorker)
+                            {
+                                if (Activity.Current != null)
```

Comment:
None of this uses ActivitySource today -- it's all manually created by App Insights and/or WebJobs.

Your comment about `AsyncLocal` is what I had originally proposed to @fabiocav (wrapped in something like `IHttpContextAccessor`... where another class can access some saved state of the current invocation). Is that something like what you were thinking?

I'm thinking about moving to that now after your comment. Any odd dependencies we take on Activity like this may be troublesome to decouple in the future. For example, if no one is listening, then stuff starts to fall apart b/c we don't have an Activity to store this state on.